File: totosuki/ABC/B/209.py
import sys
input = sys.stdin.buffer.readline

N, X = map(int, input().split())
A = list(map(int, input().split()))
answer = "Yes"

for i in range(N):
  if i % 2:
    X -= A[i] - 1
  else:
    X -= A[i]
  
  if X < 0:
    answer = "No"

print(answer)

File: OCH_Tao/ABC/B/209.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ").map(x => parseInt(x, 10));
console.log(A.reduce((a, b) => a + b, 0) - Math.floor(N / 2) <= X ? "Yes" : "No");